protos/grades_services.proto

```diff
@@ -18,19 +18,131 @@ option go_package = "github.com/BetterGR/grades-microservice/protos";
 
 package grades;
 
-// The greeting service definition.
-service Grades {
-  // Sends a greeting
-  rpc GetStudentGrade ( GradeRequest ) returns (GradeReply) {}
+//GradesService is a microservice responsible for managing grades
+service GradesService {
+  // GetCourseGrades returns all students grades enrolled in a specific course
+  rpc GetCourseGrades(GetCourseGradesRequest) returns (GetCourseGradesResponse);
+
+  // GetStudentCourseGrades returns a specific student grades in specific course
+  rpc GetStudentCourseGrades(GetStudentCourseGradesRequest) returns (GetStudentCourseGradesResponse);
+
+  // AddHomeworkGrade adds a homework grade for a student in specific course
+  rpc AddHomeworkGrade(AddHomeworkGradeRequest) returns (Empty);
+
+  // AddExamGrade adds an exam grade for a student in specific course
+  rpc AddExamGrade(AddExamGradeRequest) returns (Empty);
+
+  // UpdateHomeworkGrade updates a homework grade for a student in specific course
+  rpc UpdateHomeworkGrade(UpdateHomeworkGradeRequest) returns (Empty);
+
+  // UpdateExamGrade updates an exam grade for a student in specific course
+  rpc UpdateExamGrade(UpdateExamGradeRequest) returns (Empty);
+
+  // DeleteHomeworkGrade deletes a homework grade for a student in specific course
+  rpc DeleteHomeworkGrade(DeleteHomeworkGradeRequest) returns (Empty);
+
+  // DeleteExamGrade deletes an exam grade for a student in specific course
+  rpc DeleteExamGrade(DeleteExamGradeRequest) returns (Empty);
+}
+
+//------------------------------- Data Types -------------------------------
+// ExamGrade is a single Exam grade message
+message ExamGrade {
+  string course = 1;
+  string exam_type = 2;
+  string grade = 3;
+}
+
+// HomeworkGrade is a single Homework grade message
+message HomeworkGrade {
+  string course = 1;
+  string homework_number = 2;
+  string grade = 3;
+}
+
+// StudentCourseGrades represent all specific course grades for a student
+message StudentCourseGrades {
+  string student_id = 1;
+  string course_id = 2;
+  repeated ExamGrade exams = 3;
+  repeated HomeworkGrade homeworks = 4;
+}
+
+// StudentGrades represent all grades for a student
+message StudentGrades {
+  string student_id = 1;
+  repeated StudentCourseGrades courses = 2;
+}
+
+// Empty message could be used later to validate the request
+message Empty {}
+
+//------------------------------- Request and Response -------------------------------
```

Comment:
Same as above. If it makes sense to categorize different structs in the same package then a new file is the way to go.